### DISCUSSION TITLE
How to distribute a single topic to multiple 'consumers'

### DISCUSSION THREAD

**Discussion by benstadin** [emqx/emqx#7802] [Q&A]
The title needs some context: I need to combine MQTT which handles the client-facing communication with Kafka or an alternative which handles the server side processing at scale. 

Consider the following:
- a MQTT topic '/path/to/somedata'
- there are many MQTT writers to '/path/to/somedata'
- the data on this topic must be consumed and processed via Kafka (or some alternative). 
- Let's say we have 10 Kafka consumers and 10 partitions available for parallel processing of the MQTT topic
- Challenge: What's the most efficient way, which scales with the number of subscribers to the topic, to distribute the data to Kafka? For example in such way that the MQTT to Kafka bridge does not become a bottleneck, but rather there could be like multiple MQTT subscribers to the topic - each getting only some parts of the data?

I understand that things could be simpler if '/path/to/somedata' was actually something like '/path/to/x/somedata' where x was a differentiating factor. However, this is not possible with the data that I have.

**Comment by zmstone** [comment]
hi @benstadin 
Are you writing an MQTT consumer to bridge data from EMQX to Kafka?
You may consider using EMQX enterprise edition where Kafka bridge is built-in.

Otherwise you can try shared subscribe https://www.emqx.io/docs/en/v4.3/advanced/shared-subscriptions.html

**Reply by benckx** [reply]
Is the option "Data Bridge to Kafka" described here only available in Enterprise Edition?
https://docs.emqx.com/en/enterprise/v4.4/rule/bridge_kafka.html

I'm running EMQX 4.4.3 in a Docker container and this option isn't listed.

<img width="540" alt="Screenshot 2022-05-04 at 09 28 32" src="https://user-images.githubusercontent.com/8626080/166640748-52996f03-82aa-44d5-82f6-68b2a69eb367.png">

Edit: I managed to run EE and the option is indeed listed.

**Comment by zmstone** [comment]
coverting to a discussion.

**Comment by benstadin** [comment]
@zmstone Thanks for the swift answer. The shared subscription looks like a possible fit. Can this also be used for wildcard subscriptions? Like:

'/path/to/somedata/+'

**Reply by zmstone** [reply]
yes.
